src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -40,416 +44,554 @@
+                    totalBytesRead += bytesRead;
+                    bos.write(buffer, 0, bytesRead);
+                    bytesRead = is.read(buffer);
+                }
+                String response = new String(bos.toByteArray());
+                if (log.isDebugEnabled())
+                    log.debug("Response is " + totalBytesRead + " bytes: \n" + response);
+
+                Matcher m = thirdPartyContentPattern.matcher(response);
+                if (m.matches()) {
+                    log.debug("Response matches expected third party pattern!");
+                    bingo(Alert.RISK_MEDIUM, Alert.CONFIDENCE_MEDIUM, Constant.messages.getString("ascanbeta.insecurehttpmethod.detailed.name", HttpRequestHeader.CONNECT),
+                            Constant.messages.getString("ascanbeta.insecurehttpmethod.connect.exploitable.desc", HttpRequestHeader.CONNECT), null,
+                            null, // parameter being attacked: none.
+                            "", // attack
+                            Constant.messages.getString("ascanbeta.insecurehttpmethod.connect.exploitable.extrainfo", thirdpartyHost),
+                            Constant.messages.getString("ascanbeta.insecurehttpmethod.soln"), response, // evidence,
+                            this.getBaseMsg());
+                    return;
+                } else {
+                    log.debug("Response does *not* match expected third party pattern");
+                }
+                return;
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug(
+                            "Could not establish a socket connection to a third party using the CONNECT HTTP method: NULL socket returned, or non-200 response");
+                    log.debug("The status line returned: " + statusLine);
+                }
+                return;
+            }
+        } catch (Exception e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Could not establish a socket connection to a third party using the CONNECT HTTP method", e);
+            }
+        }finally{
+            try{is.close();}catch(Exception e){}
+            try{os.close();}catch(Exception e){}
+            try{bos.close();}catch(Exception e){}
+            try{socket.close();}catch(Exception e){}
+        }
+    }
+
+    private final void testHttpMethod(String httpMethod) throws Exception {
+        //TODO:chose a body that exploits a vulnerability
+        String randomKey = RandomStringUtils.random(15, "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789");
+        String randomValue = RandomStringUtils.random(15, "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789");
+
+        String requestBody = '"' + randomKey + "\":\"" + randomValue + '"'; //"randomKey":"randomValue"
+
+        final HttpRequestHeader requestHeader = this.getBaseMsg().getRequestHeader();
+
+
+        requestHeader.setMethod(httpMethod);
```

Comment:
This will lead to changes of the base message, we should just change the message returned by `getNewMsg()` (same happens in `testTraceOrTrack`).